backend/apps/ifc_validation/checks/signatures/check_signatures.py

```diff
@@ -0,0 +1,338 @@
+import binascii
+from dataclasses import asdict, dataclass, fields
+import datetime
+import glob
+import json
+import os
+import subprocess
+import sys
+import tempfile
+from typing import Any, List, Optional
+from typing import Tuple
+from enum import Enum, auto
+
+# @nb These (rather incomplete) bindings are no
+# longer needed, we just use the openssl executable
+# from asn1crypto import cms
+# from OpenSSL import crypto
+
+# pip install python-ranges
+from ranges import Range, RangeSet
+
+import re
+import base64
+
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding, rsa, ec
+from cryptography.exceptions import InvalidSignature
+from cryptography.x509.oid import ExtensionOID
+
+
+class VerificationResult(Enum):
+    invalid = auto()
+    valid_unknown_cert = auto()
+    valid_known_cert = auto()
+
+
+@dataclass
+class signature_data:
```

Comment:
```suggestion
class SignatureData:
```
CapitalCase for python classes.